src/Processors/Formats/Impl/ArrowColumnToCHColumn.cpp

```diff
@@ -1072,14 +1072,21 @@ Block ArrowColumnToCHColumn::arrowSchemaToCHHeader(
         auto arrow_column = std::make_shared<arrow::ChunkedArray>(array_vector);
         std::unordered_map<std::string, DictionaryInfo> dict_infos;
         bool skipped = false;
-        bool allow_null_type = false;
+        bool allow_null_type = field->nullable();
```

Comment:
It doesn't make sense. `allow_null_type` indicates if `readColumnFromArrowColumn` can return column with type `Nullable(Nothing)`, and during schema inference we don't want such type as it's incomplete.

Reply:
Updated.